### PR TITLE
feat: add algolia search functionality to docs

### LINKED ISSUE #95: feat: add markprompt to docs site
https://markprompt.com/docs#docusaurus

I'll add more context later 😀

### PR DESCRIPTION
**Pull request by OgDev-01**
## Description
The PR fixes the docs search issue with Algolia

### Notes
Didn’t see a need to add markprompt to the implementation as [algolia DocSearch](https://docsearch.algolia.com/) alone was able to fix the search. Thanks @Ant-Shell for the helpful resources you provided, it all made the fix faster👍

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
closes #95 

### Before
https://github.com/open-sauced/docs/assets/62995161/490d446e-4f90-403e-8500-30acc3fbbfb4
### After
https://github.com/open-sauced/docs/assets/62995161/596f4d0d-256b-40c9-b11a-4bda0362973b

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [x] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Review by bdougie — APPROVED** [review]
Works as expected.

**Comment by bdougie** [comment]
@BekahHW when we get you review, we can merge.

File: docusaurus.config.js [hunk 1]
```diff
@@ -1,126 +1,126 @@
...
+  projectName: "docs.opensauced.pizza", // Usually your repo name.
   trailingSlash: true,
   themeConfig: {
     algolia: {
-      appId: 'bh4d9od16a',
-      apiKey: '1691a4a5b06166984397aa30153f663c',
-      indexName: 'opensauced',
-      algoliaOptions: { 'facetFilters': ["type:$TYPE"] },
+      appId: "RH1WG3CUDK",
+      apiKey: "ea723aba7d69f2f36feac9292f2d2fe9",
```
Comment (BekahHW) [inline docusaurus.config.js:19]: @OgDev-01 just double checking, this is a public API key that's safe to expose here, right?

**Review by BekahHW — APPROVED** [review]

**Comment by OgDev-01** [comment]
Still hitting blockers trying to implement `.env` support... I'll merge this as is for now, and revisit the env addition when i have the bandwidth

**Pull request merged**